### ISSUE TITLE
KeyboardInterrupt not being raised after starting the JVM

### ISSUE THREAD

**Issue by pelson** [jpype-project/jpype#813]
In #255 there was some code to demonstrate a problem with ``KeyboardInterrupt`` not being raised after starting the JVM & then pressing ``Ctrl+C``. This can be easily reproduced with:

```
import time

import jpype as jp


jp.startJVM(
    jp.getDefaultJVMPath(),
)

try:
    while True:
        print(".", end="", flush=True)
        time.sleep(1)
except KeyboardInterrupt as e:
    print('Interrupt requested')
    raise
finally:
    print("This should be always executed", flush=True)
```

As you can see, hitting ``ctrl+c`` while this is running does not result in the while-loop being terminated, and the execution continues indefinitely. In ``JPype1<1`` the ``KeyboardInterrupt`` is honoured (i.e. the process terminates) but it doesn't actually trigger the ``KeyboardInterrupt`` (nor the ``finally`` block).

One solution to this (from #255) is to add the ``-Xrs`` flag at JVM startup (From Java help: ``-Xrs: reduce use of OS signals by Java/VM (see documentation)``). Unfortunately this flag now triggers a ``SystemError: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException`` if you try to start the JVM with the ``-Xrs`` flag.

Another solution is to turn on a special signal handler, such as ``signal.signal(signal.SIGINT, handler)`` where ``handler`` just ``raise KeyboardInterrupt()``. This solution works for JPype 0.6, 0.7 and 1.0, but it is not expected Python behaviour to have to enable ``KeyboardInterrupt``s in this way.

Is there something missing in JPype that needs to be done to enable this (expected) behaviour out of the box?

**Comment by Thrameos** [comment]
Thanks for the bug report.  I will see what I can do.   Trying to get the two virtual machines to play nicely at the signal level is challenging.   The old solution is for the entire process to get shutdown because the JVM crashed, which basically made JPype unusable in interactive sessions.   So I redirected to send interrupt to all threads but Python appears not to respect that.   I set the handlers to chain so it should have called both the Python one and the Java one, but your bug report shows that does not happen currently.

It is unlike that I can fix this until 1.1 as I may have to expose control switches to the user.

**Comment by Thrameos** [comment]
Please check if #814 solves your issue.

**Comment by pelson** [comment]
Yep, thanks @Thrameos, it now seems to correctly handle KeyboardInterrupt. The ``-Xrs`` flag still causes the ``SystemError: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException`` though - is this expected?

The change appears to be fairly tightly bound, is it considered a bug-fix, and will it therefore be in v1.0.2 or v1.1.0?

**Comment by Thrameos** [comment]
It should be in 1.0.2  hopefully released shortly.
